File: vripper-server/src/main/java/tn/mnlr/vripper/services/GlobalState.java
package tn.mnlr.vripper.services;

import lombok.Getter;

import java.util.Objects;

@Getter
public class GlobalState {

    private final long running;
    private final long queued;
    private final long remaining;
    private final long error;

    GlobalState(long running, long queued, long remaining, long error) {
        this.running = running;
        this.queued = queued;
        this.remaining = remaining;
        this.error = error;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        GlobalState that = (GlobalState) o;
        return running == that.running && queued == that.queued && remaining == that.remaining && error == that.error;
    }

    @Override
    public int hashCode() {
        return Objects.hash(running, queued, remaining, error);
    }
}

File: vripper-core/src/main/kotlin/me/mnlr/vripper/model/GlobalState.kt
package me.mnlr.vripper.model

data class GlobalState(
    val running: Int,
    val remaining: Int,
    val error: Int,
    val loggedUser: String,
    val downloadSpeed: String
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as GlobalState

        if (running != other.running) return false
        if (remaining != other.remaining) return false
        if (error != other.error) return false
        if (loggedUser != other.loggedUser) return false
        if (downloadSpeed != other.downloadSpeed) return false

        return true
    }

    override fun hashCode(): Int {
        var result = running
        result = 31 * result + remaining
        result = 31 * result + error
        result = 31 * result + loggedUser.hashCode()
        result = 31 * result + downloadSpeed.hashCode()
        return result
    }
}